src/Microsoft.ML.Data/DataLoadSave/TransformerChain.cs

```diff
@@ -36,20 +36,33 @@ public enum TransformerScope
         Everything = Training | Testing | Scoring
     }
 
+    [BestFriend]
+    internal interface ITransformerAccessor
+    {
+        ITransformer[] Transformers { get; }
+        TransformerScope[] Scopes { get; }
+    }
+
     /// <summary>
     /// A chain of transformers (possibly empty) that end with a <typeparamref name="TLastTransformer"/>.
     /// For an empty chain, <typeparamref name="TLastTransformer"/> is always <see cref="ITransformer"/>.
     /// </summary>
-    public sealed class TransformerChain<TLastTransformer> : ITransformer, ICanSaveModel, IEnumerable<ITransformer>
+    public sealed class TransformerChain<TLastTransformer> : ITransformer, ICanSaveModel, IEnumerable<ITransformer>, ITransformerAccessor
     where TLastTransformer : class, ITransformer
     {
-        private readonly ITransformer[] _transformers;
-        private readonly TransformerScope[] _scopes;
+        [BestFriend]
+        internal readonly ITransformer[] Transformers;
```

Comment:
>internal readonly ITransformer[] Transformers; [](start = 8, length = 46)

Didn't you introduce the internal interface so you wouldn't have to expose these things?